store/auth.ts

```diff
@@ -1,97 +1,50 @@
-    isLoading: boolean
-    error: null
-    authInfo: null
-  }
-  | {
-    kind: 'ERRORED'
-    isLoading: false
-    error: ErrorApiBase
-    authInfo: null
-  }
-  | {
-    kind: 'AUTHENTICATED'
-    isLoading: false
-    error: null
-    authInfo: {
-      userId: string
-      status: string
-      roles: {
-        archived?: boolean
-        member?: boolean
-      }
-    }
+export const useAuthStore = defineStore('auth', () => {
+  const config = getConfig()
+  const user = ref()
+  const isLoading = ref(false)
+  const token = useCookie(`goals_token_${config.ENV}`, {
+    maxAge: 60 * 60 * 24 * 30, // 30 days 
+    sameSite: 'lax'
+  })
+
+  const setToken = (newToken: string) => {
+    token.value = newToken
   }
 
-export const useAuthStore = defineStore({
-  id: 'auth-store',
-  state: (): AuthStoreState => ({
-    kind: 'UNAUTHENTICATED',
-    isLoading: false,
-    error: null,
-    authInfo: null
-  }),
-  actions: {
-    async fetchSelf () {
-      if (this.isLoading) return
+  const setUser = (user: any) => {
+    user.value = user
+  }
 
-      this.isLoading = true
+  const signin = async () => {
+    try {
+      isLoading.value = true
+      const [selfData, credentials] = await Promise.all([authAdapter.getSelf(), authAdapter.getGoalApiCredentials()])
 
-      try {
-        const { data, error } = await authAdapter.getSelf()
-        if (error != null) {
-          this.$patch({
-            kind: 'ERRORED',
-            error,
-            isLoading: false,
-            authInfo: null
-          })
-        } else if (data != null) {
-          this.$patch({
-            kind: 'AUTHENTICATED',
-            authInfo: data.info,
-            isLoading: false,
-            error: null
-          })
-          userRepo.save(data.user)
-        }
-      } catch (e) {
-        console.error(e)
-      }
+      user.value = selfData
+      token.value = credentials.token.access
+      isLoading.value = false
+    } catch (error) {
+      user.value = undefined
+      token.value = undefined
+      isLoading.value = false
     }
-  },
-  getters: {
-    getAuthState (state): AuthState {
-      if (state.kind === 'UNAUTHENTICATED') {
-        return {
-          kind: 'UNAUTHENTICATED',
-          isLoading: state.isLoading
-        }
-      } else if (state.kind === 'ERRORED') {
-        return {
-          kind: 'UNAUTHENTICATED',
-          isLoading: false,
-          error: state.error
-        }
-      } else {
-        const user = userRepo.find(state.authInfo.userId)
-        if (user == null) throw new Error('User not found in repository')
+  }
 
-        return {
-          kind: 'AUTHENTICATED',
-          isLoading: false,
-          roles: state.authInfo.roles,
-          user
-        }
-      }
-    }
+  const signout = () => {
+    token.value = undefined
+    user.value = undefined
+  }
+
+  return {
+    user,
+    token,
+    setToken,
+    setUser,
+    signin,
+    signout,
+    isLoading,
```

Comment:
Should token, setToken, setUser be exported? Or they are exported for tests?

Reply:
Thanks for the catch. Will address it.